x-pack/plugins/observability_solution/observability_ai_assistant_app/public/components/chat/chat_body.tsx

```diff
@@ -341,32 +343,69 @@ export function ChatBody({
-                <WelcomeMessage connectors={connectors} knowledgeBase={knowledgeBase} />
-              ) : (
-                <ChatTimeline
-                  messages={messages}
+                <WelcomeMessage
+                  connectors={connectors}
                   knowledgeBase={knowledgeBase}
-                  chatService={chatService}
-                  currentUser={currentUser}
-                  chatState={state}
-                  hasConnector={!!connectors.connectors?.length}
-                  onEdit={(editedMessage, newMessage) => {
-                    setStickToBottom(true);
-                    const indexOf = messages.indexOf(editedMessage);
-                    next(messages.slice(0, indexOf).concat(newMessage));
-                  }}
-                  onFeedback={handleFeedback}
-                  onRegenerate={(message) => {
-                    next(reverseToLastUserMessage(messages, message));
-                  }}
-                  onSendTelemetry={(eventWithPayload) =>
-                    chatService.sendAnalyticsEvent(eventWithPayload)
+                  onSelectPrompt={(message) =>
+                    next(
+                      messages.concat([
+                        {
+                          '@timestamp': new Date().toISOString(),
+                          message: { content: message, role: MessageRole.User },
+                        },
+                      ])
+                    )
                   }
-                  onStopGenerating={() => {
-                    stop();
-                  }}
-                  onActionClick={handleActionClick}
                 />
+              ) : (
+                <>
+                  <ChatTimeline
+                    messages={messages}
+                    knowledgeBase={knowledgeBase}
+                    chatService={chatService}
+                    currentUser={currentUser}
+                    chatState={state}
+                    hasConnector={!!connectors.connectors?.length}
+                    onEdit={(editedMessage, newMessage) => {
+                      setStickToBottom(true);
+                      const indexOf = messages.indexOf(editedMessage);
+                      next(messages.slice(0, indexOf).concat(newMessage));
+                    }}
+                    onFeedback={handleFeedback}
+                    onRegenerate={(message) => {
+                      next(reverseToLastUserMessage(messages, message));
+                    }}
+                    onSendTelemetry={(eventWithPayload) =>
+                      chatService.sendAnalyticsEvent(eventWithPayload)
+                    }
+                    onStopGenerating={() => {
+                      stop();
+                    }}
+                    onActionClick={handleActionClick}
+                  />
+
+                  {!isLoading ? (
+                    <>
+                      <StarterPrompts
+                        userPrompts={messages
```

Comment:
It's possible that user interaction causes further calls to `setScreenContext` with new suggestions. 

Example:
- User is on page A, page A added a starterPrompt: 'Where do I see X?'
- Assistant answers: 'It's possible to see that by adjusting setting Y on the page. Do you want me to do that for you?'
- User confirms
- Setting Y being set triggers a `useEffect` (or whatever) which calls `setScreenContext` again with a follow up starter prompt: 'How do I investigate X for errors?'

This may or may not be too difficult to use in practice for Kibana engineers to set up. 

The primary advantage of this approach I see is speed. Downside is it expects a lot of Kibana engineers to make it work well.

We could also ask the LLM for further suggestions but that would require a summarization, and would be (at least now) slow.

Reply:
Even in that case, I think it's hard to give good contextual prompts if the user has already started the conversation. In this example, how does an engineer know the difference between a conversation where that prompt is useful and where it is completely useless? I think we can remove some complexity here by not showing the prompts if the user has already started a conversation. This is also what ChatGPT does.